### DISCUSSION TITLE
How to Achieve Relative Binding in a Border Style

### DISCUSSION THREAD

**Discussion by wazik** [AvaloniaUI/Avalonia#16841] [Q&A]
How can I achieve relative binding within a Border style, referencing properties like Bounds.Width and Bounds.Height, without using named elements? I'm looking for a way to keep the Style reusable and not tied to a specific Name.

```
<Border Background="Aqua" CornerRadius="20 0 0 20" HorizontalAlignment="Left" Margin="5 20 0 0" Padding="10 3 30 3">
    <Border.Clip>
        <PathGeometry>
            <PathGeometry.Figures>
                <PathFigures>
                    <PathFigure StartPoint="0,0">

                        <LineSegment>
                            <LineSegment.Point>
                                <MultiBinding Converter="{StaticResource LableClipConverterA}">
                                    <Binding Path="#border.Bounds.Width" />
                                    <Binding Path="#border.Bounds.Height" />
                                </MultiBinding>
                            </LineSegment.Point>
                        </LineSegment>

                        <LineSegment>
                            <LineSegment.Point>
                                <MultiBinding Converter="{StaticResource LableClipConverterB}">
                                    <Binding Path="#border.Bounds.Width" />
                                    <Binding Path="#border.Bounds.Height" />
                                </MultiBinding>
                            </LineSegment.Point>
                        </LineSegment>

                        <LineSegment>
                            <LineSegment.Point>
                                <MultiBinding Converter="{StaticResource LableClipConverterC}">
                                    <Binding Path="#border.Bounds.Width" />
                                    <Binding Path="#border.Bounds.Height" />
                                </MultiBinding>
                            </LineSegment.Point>
                        </LineSegment>

                        <LineSegment Point="0,0"/>
                    </PathFigure>
                </PathFigures>
            </PathGeometry.Figures>
        </PathGeometry>
    </Border.Clip>

    <TextBlock Text="{Binding $parent[Border].Bounds}"/>
</Border>
```

![Screenshot 2024-08-28 113621](https://github.com/user-attachments/assets/6a6fbe44-fa00-4d48-91ef-d71bf3e7d5d8)

As part of my tests, I tried the following:
```<LineSegment>
  <LineSegment.Point>
    <MultiBinding Converter="{StaticResource LableClipConverterA}">
      <Binding Path="$self" />
      <Binding Path="$parent" />
    </MultiBinding>
  </LineSegment.Point>
</LineSegment>
```

My converter received the expected Border and StackPanel objects, but I encountered an issue when I attempted to use:
`<Binding Path="$self.Bounds" />`

It seems that $self resolves to an Avalon.Data.MultiBinding object instead of the expected Border. Could someone explain why this happens and how I might address this issue?

Any advice or alternative approaches would be greatly appreciated!

Thank you in advance for your assistance.

**Comment by IanRawley** [comment]
Which version of Avalonia are you using?  $Self in compiled multibindings was supposedly fixed in 11.1 #14935

**Comment by wazik** [comment]
Thanks for the answer. My version is 11.0.10, so that might be the reason. I'll test it and let you know the results.

However, I've noticed an inconsistency even with normal bindings. This code:
`<LineSegment Point="{Binding $self, Converter={StaticResource LabelClipConverterA}}"/>`
passes the Border as the converter value, but this code:
`<LineSegment Point="{Binding $self.Bounds.Width, Converter={StaticResource LabelClipConverterA}}"/>`
cannot compile because of the following error:

"Unable to resolve property or method of name 'Bounds' on type 'Avalonia.Media.LineSegment'."
